tests/MongoDB.Bson.Tests/Specifications/bson-corpus/BsonCorpusTestRunner.cs

```diff
@@ -35,6 +35,33 @@ public void RunTestDefinition(JsonDrivenTestCase testCase)
         {
             var test = testCase.Test;
             var testType = test["type"].AsString;
+            var shared = testCase.Shared;
+
+            JsonDrivenHelper.EnsureAllFieldsAreValid(
+                shared,
+                "_path",
+                "description",
+                "bson_type",
+                "test_key",
+                "deprecated",
+                "valid",
+                "decodeErrors",
+                "parseErrors");
+
+            if (shared.Contains("bson_type"))
+            {
+                // This field is gnored
+            }
+
+            if (shared.Contains("test_key"))
+            {
+                // This field is gnored
+            }
+
+            if (shared.Contains("deprecated"))
+            {
+                // This field is gnored
+            }
```

Comment:
These if statements seem unnecessary. The same information could be conveyed to the reader in a comment:

    JsonDrivenHelper.EnsureAllFieldsAreValid(
        shared,
        "_path",
        "description",
        "bson_type", // ignored
        "test_key", // ignored
        "deprecated", // ignored
        "valid",
        "decodeErrors",
        "parseErrors");

Reply:
Done.